semantic-core/src/Analysis/Typecheck.hs

```diff
@@ -173,7 +174,12 @@ typecheckingAnalysis = Analysis{..}
         asBool b = unify (Term Bool) b >> pure True <|> pure False
         string _ = pure (Term String)
         asString s = unify (Term String) s $> mempty
-        record fields = pure (Term (Record (Map.fromList fields)))
+        record fields = do
+          fields' <- for fields $ \ (k, v) -> do
+            addr <- alloc k
+            (k, v) <$ assign addr v
+          -- FIXME: should records reference types by address instead?
```

Comment:
I think this varies based on language? Not sure, though.

Reply:
Shouldn’t for the types, but might for values.